### DISCUSSION TITLE
Circuit without inputs

### DISCUSSION THREAD

**Discussion by sethahrenbach** [zkonduit/ezkl#133] [Q&A] [answered]
If I understand correctly, each ZK-SNARK is built from a model ONNX and JSON input. Is there a way to produce a circuit based on just the model ONNX, so that it can receive arbitrary inputs of the correct type?

**Comment by alexander-camuto** [comment, marked as answer]
The `prove` command currently builds the circuit, based on the model ONNX, then generates a proof from the resulting circuit. However we currently don't have a command for serializing the circuit to disk so that the compiled circuit can be re-used. 

Currently, the best way to achieve the effect you're looking for is to simply run `prove` where the ONNX file is constant but `input.json` differs -- the intermediate circuit that is built should be the same every time but with each time the arbitrary inputs being assigned to the "input columns" of the circuit. This would be roughly equivalent to the pipeline you're looking for.

**Reply by sethahrenbach** [reply]
Thank you, I'll play around with that
